apps/docs.blocksense.network/components/common/Drawer.tsx

```diff
@@ -0,0 +1,195 @@
+'use client';
+
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  MouseEvent,
+  useCallback,
+  ReactNode,
+  ReactElement,
+} from 'react';
+
+import { cn } from '@/lib/utils';
+
+interface DrawerContextValue {
+  isOpen: boolean;
+  toggleOpen: () => void;
+}
+
+const DrawerContext = createContext<DrawerContextValue>(
+  {} as DrawerContextValue,
+);
+
+export const Drawer = ({
+  children,
+  open,
+  onOpenChange,
+}: {
+  children: ReactNode;
+  open?: boolean;
+  onOpenChange?: (open: boolean) => void;
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(open ?? false);
+
+  useEffect(() => {
+    if (open !== undefined) {
+      setIsOpen(open);
+    }
+  }, [open]);
+
+  const handleClickOutside = useCallback((event: Event) => {
+    const drawerElement = document.querySelector('.drawer__content');
+    if (drawerElement && !drawerElement.contains(event.target as Node)) {
+      handleClose();
+    }
+  }, []);
+
+  const handleKeyDown = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      handleClose();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isOpen) return;
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen, handleClickOutside, handleKeyDown]);
+
+  const handleClose = () => {
+    setIsOpen(false);
+    onOpenChange?.(false);
+  };
+
+  const handleOpen = () => {
+    setIsOpen(true);
+    onOpenChange?.(true);
+  };
+
+  return (
+    <DrawerContext.Provider value={{ isOpen, toggleOpen: handleOpen }}>
+      {children}
+    </DrawerContext.Provider>
+  );
+};
+
+export const DrawerTrigger = ({
+  children,
+  asChild,
+}: {
+  children: ReactNode;
+  asChild?: boolean;
+}) => {
+  const { toggleOpen } = useContext(DrawerContext) ?? {};
+
+  if (!toggleOpen) {
+    throw new Error('DrawerTrigger must be used within a Drawer.');
+  }
```

Comment:
```suggestion
  const { toggleOpen } = useContext(DrawerContext);
```

You can skip the if for `toggleOpen` and `?? {}` everywhere